### PR TITLE
Message: add `References` and `In-Reply-To` headers shortcuts

### PR DESCRIPTION
**Pull request by wujku**

### CODE REVIEW

File: src/Message/AbstractMessage.php [hunk 1]
```diff
@@ -124,6 +124,22 @@
     }
 
     /**
+     * Get message references (from headers).
+     *
+     * @return array
+     */
+    final public function getReferences()
```
Comment (Slamdunk) [inline src/Message/AbstractMessage.php]: Add `array` return type

File: src/Message/AbstractMessage.php [hunk 2]
```diff
@@ -124,6 +124,22 @@
...
+    {
+        $references = $this->getHeaders()->get('references');
+
+        if (\is_array($references)) {
+            return \array_unique($references);
```
Comment (Slamdunk) [inline src/Message/AbstractMessage.php]: I don't like it very much to alter a header content. Emailing is a brittle system full of exceptions: even though it seems harmless, I would prefer to let the user do it if needed.

File: src/Message/AbstractMessage.php [hunk 3]
```diff
@@ -124,6 +124,22 @@
...
+        }
+
+        return [];
```
Comment (Slamdunk) [inline src/Message/AbstractMessage.php]: We can use a single return with ternary operator:
```php
return \is_array($references) ? $references : [];
```

File: tests/MessageTest.php [hunk 4]
```diff
@@ -594,4 +594,21 @@ public function testMultipartMessageWithoutCharset()
...
     }
+
+    public function testGetReferences()
+    {
+        $fixture = $this->getFixture('references');
+        $this->mailbox->addMessage($fixture);
+
+        $message = $this->mailbox->getMessage(1);
+
+        $this->assertSame(2, \count($message->getReferences()));
```
Comment (Slamdunk) [inline tests/MessageTest.php]: `assertCount`

File: tests/fixtures/references.eml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+Message-ID: <123@example.com>
+From: no_host
+Cc: "This one: is \"right\"" <ding@dong.com>, No-address
+References: <231d9ac57aec7d8c1a0eacfeab8af6f3@example.com> <08F04024-A5B3-4FDE-BF2C-6710DE97D8D9@example.com>
```
Comment (Slamdunk) [inline tests/fixtures/references.eml]: Would you be so kind to also add the same API for `In-Reply-To` header?
https://tools.ietf.org/html/rfc4021#section-2.1.9
https://tools.ietf.org/html/rfc2822#section-3.6.4
It seems to have exactly the same behaviour.

File: tests/MessageTest.php [hunk 6]
```diff
@@ -594,4 +594,21 @@ public function testMultipartMessageWithoutCharset()
...
+
+        $fixture = $this->getFixture('plain_only');
+        $this->mailbox->addMessage($fixture);
+
+        $message = $this->mailbox->getMessage(2);
+
+        $this->assertSame(0, \count($message->getReferences()));
```
Comment (Slamdunk) [inline tests/MessageTest.php]: `assertCount`

File: tests/fixtures/references.eml (hunk 5 above)
Comment (wujku) [reply tests/fixtures/references.eml]: Check this 2b4f5af

File: tests/MessageTest.php [hunk 7]
```diff
@@ -594,4 +594,38 @@ public function testMultipartMessageWithoutCharset()
...
+    }
+
+    public function testGetReferences()
+    {
+        $fixture = $this->getFixture('references');
+        $this->mailbox->addMessage($fixture);
+
+        $message = $this->mailbox->getMessage(1);
+
+        $this->assertCount(2, $message->getReferences());
```
Comment (Slamdunk) [inline tests/MessageTest.php]: Please add also specific assertion to the content of the referencies, e.g. `$this->assertContains('<231d9ac57aec7d8c1a0eacfeab8af6f3@example.com>, $message->getReferences());'`

File: src/Message/AbstractMessage.php [hunk 8]
```diff
@@ -124,6 +124,30 @@
     }
 
     /**
+     * Get message In-Reply-To (from headers).
+     *
+     * @return array
+     */
+    final public function getInReplyTo(): array
```
Comment (Slamdunk) [inline src/Message/AbstractMessage.php]: Please add these two new signatureto `Message\BasicMessageInterface`

**Comment by Slamdunk** [comment]
Thank you @wujku

**Pull request merged**